compiler/rustc_infer/src/traits/project.rs

```diff
@@ -90,6 +90,7 @@ impl ProjectionCacheKey<'tcx> {
 pub enum ProjectionCacheEntry<'tcx> {
     InProgress,
     Ambiguous,
+    Recur,
```

Comment:
```suggestion
    Recursive,
```
I had to read the doc comment on the `recur` method to understand what this meant.